packages/app/src/cli/models/extensions/specifications/payments_app_extension_schemas/base_payments_app_extension_schema.ts

```diff
@@ -12,6 +12,7 @@ export interface BasePaymentsAppExtensionDeployConfigType {
   merchant_label: string
   supported_countries: string[]
   supported_payment_methods: string[]
+  supported_buyer_contexts?: {currency: string; countries?: [string, ...string[]]}[]
```

Comment:
Am I understanding correctly that this type definition enforces the non-empty condition? (`countries?: [string, ...string[]]` versus a simpler `string[]`)

If so, do we also need the `nonempty()` check at line 70? I guess it doesn't hurt either 🤷

Reply:
If we want to make `supported_buyer_contexts` required at some point later, are we able to do so or is making it nullable today forcing lifetime support of a nullable value?

Reply:
> Am I understanding correctly that this type definition enforces the non-empty condition? (`countries?: [string, ...string[]]` versus a simpler `string[]`)

Correct!

> If so, do we also need the `nonempty()` check at line 70? I guess it doesn't hurt either 🤷

We need both. `SupportedBuyerContextsSchema` is the schema definition that will validate and parse input (by calling `#parse` on it). `BasePaymentsAppExtensionDeployConfigType` is just the type definition and the output of the schema. We can compute one from the type from the schema (`type SchemaType  = zod.infer<typeof Schema>`) but there must be a reason we don't!

> If we want to make `supported_buyer_contexts` required at some point later, are we able to do so or is making it nullable today forcing lifetime support of a nullable value?

I don't think we have the concept of config schema versions right now, so we'd likely have to introduce a new pattern to support this. It may already be used by other parts of the CLI but we're not making an irreversible decision.